subsys/bluetooth/host/conn_internal.h

```diff
@@ -269,12 +262,72 @@ struct bt_conn {
 		uint16_t subversion;
 	} rv;
 #endif
+
+	/* For ACL: List of data-ready L2 channels. Used by TX processor for
+	 * pulling fragments. Channels are only ever removed from this list when
+	 * a whole PDU (ie all its frags) have been sent.
+	 *
+	 * For ISO: Non-null to signal data has something to send.
+	 */
+	sys_slist_t		upper_data_ready;
+
+	/* Node for putting this connection in a data-ready mode for the bt_dev.
+	 * This will be used by the TX processor to then fetch HCI frags from it.
+	 */
+	sys_snode_t		_conn_ready;
+	atomic_t		_conn_ready_lock;
+
+	/* Holds the number of packets that have been sent to the controller but
+	 * not yet ACKd (by receiving an Number of Completed Packets). This
+	 * variable can be used for deriving a QoS or waterlevel scheme in order
+	 * to maximize throughput/latency.
+	 * It's an optimization so we don't chase `tx_pending` all the time.
+	 */
+	atomic_t		in_ll;
```

Comment:
How does this differ from `tx_pending`, which is defined as 
```
/* Pending TX that are awaiting the NCP event. len(tx_pending) == in_ll */
```

Reply:
tx_pending is a list. This is an optimization to not have to traverse the list to know its length for every single packet we send.

Reply:
note that I'm fine with removing `in_ll` and introducing it in a later PR if that's what you want. Just reply what you prefer :)

Reply:
That's makes sense :) 

We do have `sys_slist_len` to calculate this (and I guess we could eventually add a cache to the slist struct later if we wanted). 

Either way is OK